### PR TITLE
Prometheus metrics & mixin

### PR DESCRIPTION
**Pull request by kskewes**
Rebased Prometheus PR on master.

Closes: #177 

- add http handler on `/metrics`. 
  There aren't any sensitive endpoints on this port, but if metrics are considered sensitive for some then I can move to a custom port.
- add standard `build_info` and `go_build_info` metrics
- add custom unseal error & request metrics - RE(D)
- add mixin with:
  - Grafana dashboard
  - Alerts
- add `PodMonitor` jsonnet for those using Prometheus Operator (us)

Additional thoughts:
- `make test` is currently failing for me on master as well as the `prometheus` branch: 
  `--- FAIL: TestHttpCert (1.37s)    `
- I did the bulk of this before key rotation was done and didn't experience any errors due to prometheus integration. 
- Key rotation metrics/dashboard/alert could be easily added in a subsequent PR.

Testing done:
- with `kind`
- create sealed secret - observe request counter increment
- submit SS with new secret - observe request counter increment
- submit SS with modified namespace (incorrect) - observe error increment
- submit SS with modified encrypted value (incorrect) - observe error increment
- delete secret - observe request counter increment
- delete SS - observe request counter increment

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -135,6 +135,8 @@ controller.yaml: controller.jsonnet controller-norbac.jsonnet
 
 controller-norbac.yaml: controller-norbac.jsonnet
 
+controller-podmonitor.yaml: controller.jsonnet controller.image controller-norbac.jsonnet
```
Comment (mkmik) [inline Makefile]: as of #367  (multi-arch containers) the *.yaml rules no longer depend on the controller.image

```suggestion
controller-podmonitor.yaml: controller.jsonnet controller-norbac.jsonnet
```

File: prometheus-mixin/Makefile [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Prometheus Mixin Makefile
+# Heavily copied from upstream project kubenetes-mixin
+
+# TODO: Find out why official prom images segfault
+# PROMETHEUS_IMAGE := prom/prometheus:v2.1.0
+PROMETHEUS_IMAGE := registry.gitlab.com/kskewes/docker-promtool:latest
+
+JSONNET_FMT := jsonnetfmt -n 2 --max-blank-lines 2 --string-style s --comment-style s
```
Comment (mkmik) [inline prometheus-mixin/Makefile]: until `jsonnetfmt` learns to read its config from a `.jsonnetfmtrc` file in the project directory, I'm against using non-standard formatting; it just makes life harder for those people who have jsonnet autoformatters setup in their editor of choice.

File: prometheus-mixin/Makefile [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# Prometheus Mixin Makefile
```
Comment (mkmik) [inline prometheus-mixin/Makefile]: let's put `prometheus-mixin` in a `contrib` top level directory

Comment (mkmik) [reply prometheus-mixin/Makefile]: also please add a README.md file in this directory explaining what it is

**Comment by mkmik** [comment]
unfortunately the tests fail because of the way multiarch docker images are currently built.

the official docker ecosystem is often very automation unfriendly; there are alternative tools that will allow us to build images and manifests offline without having to push them to a registry; I'll try to play with that and regain the ability to run integration tests in unprivileged PRs

**Comment by mkmik** [comment]
@kskewes,  I'm happy to merge this and apply further cleanups myself. Please clarify why you marked this PR as "WIP", do you plan pushing more changes later or it's just to signal that it's not "done" done?

**Comment by kskewes** [comment]
> @kskewes, I'm happy to merge this and apply further cleanups myself. Please clarify why you marked this PR as "WIP", do you plan pushing more changes later or it's just to signal that it's not "done" done?

Just to signal it's not done done. Thanks for the quick feedback.
I've started tending to it and will submit this weekend.

File: prometheus-mixin/Makefile (hunk 2 above)
Comment (kskewes) [reply prometheus-mixin/Makefile]: Sure. Looks like these settings have been propagating around the various mixin projects, see: [prometheus mixin](https://github.com/prometheus/prometheus/blob/master/documentation/prometheus-mixin/Makefile#L1).

File: prometheus-mixin/Makefile (hunk 3 above)
Comment (kskewes) [reply prometheus-mixin/Makefile]: Sure, updated.

File: Makefile (hunk 1 above)
Comment (kskewes) [reply Makefile]: Ah, missed this in rebase, done now.

**Pull request merged**